src/client/datascience/baseJupyterSession.ts

```diff
@@ -110,7 +110,9 @@ export abstract class BaseJupyterSession implements IJupyterSession {
                 await this.shutdownSession(this.session, this.statusHandler, false);
                 traceVerbose('Shutdown session - get restart session');
                 if (this.restartSessionPromise) {
-                    const restartSession = await this.restartSessionPromise;
+                    this.restartSessionPromise.token.cancel();
```

Comment:
Now we have cancellation tokens to kill restart sessions much faster (previously we'd have to wait for that to complete)

Reply:
THis is just a by product, i'm passing cancellation tokens when starting kernels.
Basically kernel.start & notebook.connect method are called in 3 places & all will have a cancellation token & that makes things better